services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java

```diff
@@ -638,10 +639,17 @@ public static Stream<ClientScopeModel> getRequestedClientScopes(KeycloakSession
             return clientScopes;
         }
 
-        // skip scopes that were explicitly requested using the dynamic scope format
+        // skip organization-related scopes that were explicitly requested using the dynamic scope format
         // we don't want dynamic and default client scopes duplicated
-        // always include the dedicated client scope that maps to the client itself
-        clientScopes = clientScopes.filter(scope -> scope.equals(client) || !scopeParam.contains(scope.getName() + ClientScopeModel.VALUE_SEPARATOR));
+        clientScopes = clientScopes.filter(scope -> {
+            List<ProtocolMapperModel> orgMappers = scope.getProtocolMapperByType(OrganizationMembershipMapper.PROVIDER_ID);
+
+            if (!orgMappers.isEmpty()) {
+                return !scopeParam.contains(scope.getName() + ClientScopeModel.VALUE_SEPARATOR);
+            }
+
+            return true;
+        });
```

Comment:
I moved the `scope.getProtocolMapperByType()` invocation to avoid creating a list and a possible lookup, as `!scopeParam.contains()` should be cheaper to evaluate.
I also noticed `scope.equals(client)` has been removed in this PR. Just want to double-check if it's intentional. 

```suggestion
        clientScopes = clientScopes.filter(scope ->
                !scopeParam.contains(scope.getName() + ClientScopeModel.VALUE_SEPARATOR) ||
                        scope.getProtocolMapperByType(OrganizationMembershipMapper.PROVIDER_ID).isEmpty()
        );
```

Reply:
The fix is to only check the requested params if there are mappers with that type.